cli/command_snapshot_create.go

```diff
@@ -164,7 +186,7 @@ func startTimeAfterEndTime(startTime, endTime time.Time) bool {
 		startTime.After(endTime)
 }
 
-func snapshotSingleSource(ctx context.Context, rep repo.RepositoryWriter, u *snapshotfs.Uploader, sourceInfo snapshot.SourceInfo) error {
+func snapshotSingleSource(ctx context.Context, rep repo.RepositoryWriter, u *snapshotfs.Uploader, sourceInfo snapshot.SourceInfo, tags map[string]string) error {
```

Comment:
The tags should be validated before starting the upload, in particular so the tag names (keys) do not conflict with the labels used to identify snapshots (manifest type, hostname, user, path, etc). See comment below about storing the tags as manifest labels.

Somewhere in this function seems like a good place to do it.